lldb/source/Target/Process.cpp

```diff
@@ -1064,6 +1065,29 @@ const char *Process::GetExitDescription() {
 bool Process::SetExitStatus(int status, llvm::StringRef exit_string) {
   // Use a mutex to protect setting the exit status.
   std::lock_guard<std::mutex> guard(m_exit_status_mutex);
+  telemetry::ScopedDispatcher<telemetry::ProcessExitInfo> helper;
+
+  // Find the executable-module's UUID, if available.
+  UUID exec_uuid;
+  // Check if there is (still) a valid target and get the debugger and exec_uuid
+  // from it.
+  TargetSP target_sp(Debugger::FindTargetWithProcessID(m_pid));
+  if (target_sp) {
+    helper.SetDebugger(&(target_sp->GetDebugger()));
+    exec_uuid = target_sp->GetExecModuleUUID();
```

Comment:
We save the Target's exec-module's UUID in the Target object because the Module object is gone by the time we're here.

Reply:
I can't seem to reproduce it now but I was getting a crash (due to accessing the ModuleSP object here), hence the check on the TargetSP and the code to save it.

But anyway,  i'll revert that back to the simple case. We can fix the crash if/when it manifest again

Reply:
@labath  Ah, I've found the test that crashed! (See the build log on this PR). Without the check for the validity of the Target pointer and the module pointer, the test crashed/timedout. By contrast, if we added this check, the whole test suite passed:

```
  TargetSP target_sp(Debugger::FindTargetWithProcessID(m_pid));
   if (target_sp) {
    helper.SetDebugger(&(target_sp->GetDebugger()));
    if (ModuleSP mod = target_sp->GetExecutableModule())
      module_uuid = mod->GetUUID();
   }

```

Reply:
That sort of makes sense. The target would not be available if it was already being destroyed. However, `Debugger::FindTargetWithProcessID` is still a very roundabout way to check for target validity. You should be able to achieve the same thing with:
```
if (TargetSP target_sp = m_target_wp.lock()) {
  ...
}
```